src/AggregateFunctions/UniquesHashSet.h

```diff
@@ -424,14 +424,30 @@ class UniquesHashSet : private HashTableAllocatorWithStackMemory<(1ULL << UNIQUE
 
         alloc(new_size_degree);
 
-        for (size_t i = 0; i < m_size; ++i)
+        if (m_size <= 1)
         {
-            HashValue x = 0;
-            DB::readIntBinary(x, rb);
-            if (x == 0)
-                has_zero = true;
-            else
-                reinsertImpl(x);
+            for (size_t i = 0; i < m_size; ++i)
+            {
+                HashValue x = 0;
+                DB::readIntBinary(x, rb);
+                if (x == 0)
+                    has_zero = true;
+                else
+                    reinsertImpl(x);
+            }
+        }
+        else
+        {
+            auto hs = std::make_unique<HashValue[]>(m_size);
+            rb.readStrict(reinterpret_cast<char *>(hs.get()), m_size * sizeof(HashValue));
+
+            for (size_t i = 0; i < m_size; ++i)
```

Comment:
I wouldn't expect reading one value to often be the case and readIntBinary does the same rb.readStrict under the hood (yes, with constant size, but let's neglect this). so, maybe just always use this loop?

Reply:
I've tested it and it does have an impact. Not having this makes reading reading statuses with just one value around 25-30% slower. I'll add a benchmark.

Reply:
ok